master/internal/api_experiment.go

```diff
@@ -649,3 +652,59 @@ func (a *apiServer) MetricBatches(req *apiv1.MetricBatchesRequest,
 		}
 	}
 }
+
+func (a *apiServer) TrialsSnapshot(req *apiv1.TrialsSnapshotRequest,
+	resp apiv1.Determined_TrialsSnapshotServer) error {
+	experimentID := int(req.ExperimentId)
+	batchesProcessed := int(req.BatchesProcessed)
+	metricName := req.MetricName
+	metricType := req.MetricType
+	if metricType == apiv1.MetricType_METRIC_TYPE_UNSPECIFIED {
+		return errors.New("must specify a metric type")
+	}
+	if metricName == "" {
+		return errors.New("must provide metric name")
+	}
+
+	var startTime time.Time
+	for {
+		var response apiv1.TrialsSnapshotResponse
+
+		var newTrials []*apiv1.TrialsSnapshotResponse_Trial
+		var endTime time.Time
+		var err error
+		switch metricType {
+		case apiv1.MetricType_METRIC_TYPE_TRAINING:
+			newTrials, endTime, err = a.m.db.TrainingTrialsSnapshot(experimentID,
+				batchesProcessed, metricName, startTime)
+		case apiv1.MetricType_METRIC_TYPE_VALIDATION:
+			newTrials, endTime, err = a.m.db.ValidationTrialsSnapshot(experimentID,
```

Comment:
comment: I started thinking about this in depth and I see why you wanted the timestamp calculated in postgres for all this time stuff.